src/utils/load-tsconfig-options.js

```diff
@@ -0,0 +1,137 @@
+const ts = require('typescript');
+const { join, dirname, resolve } = require('path');
+const fs = require('fs');
+const { paramCase } = require('param-case');
+
+/**
+ * @typedef {object} LoadTsconfigInit
+ * @property {string} base
+ * @property {string} start
+ * @property {string} filename
+ */
+
+/**
+ * @description Adapted from https://github.com/vercel/vercel/blob/18bec983aefbe2a77bd14eda6fca59ff7e956d8b/packages/build-utils/src/fs/run-user-scripts.ts#L289-L310
+ * @param {LoadTsconfigInit}
+ * @returns {string | null}
+ */
+function walkParentDirs({ base, start, filename }) {
+  let parent = '';
+
+  for (let current = start; base.length <= current.length; current = parent) {
+    const fullPath = join(current, filename);
+
+    if (fs.existsSync(fullPath)) {
+      return fullPath;
+    }
+
+    parent = dirname(current);
+  }
+
+  return null;
+}
+
+/**
+ * @param {ts.CompilerOptions} options
+ * @param {string | undefined} key
+ * @param {(value: string) => string} [callback]
+ * @returns {string | undefined}
+ */
+function convertEnumCompilerOptions(enumCompilerOptions, key, callback) {
+  if (key == null) {
+    return undefined;
+  }
+  const value = enumCompilerOptions[key];
+  return typeof callback === 'function' ? callback(value) : value;
+}
+
+/**
+ * @param {string} value
+ * @returns {string}
+ */
+function toLowerCase(value) {
+  return value.toLowerCase();
+}
+
+/**
+ * @param {ts.NewLineKind} newLine
+ * @returns {string | undefined}
+ */
+function normalizeNewLineOption(newLine) {
+  switch (newLine) {
+    case ts.NewLineKind.CarriageReturnLineFeed:
+      return 'crlf';
+    case ts.NewLineKind.LineFeed:
+      return 'lf';
+    default:
+      return undefined;
+  }
+}
+
+/**
+ * @param {ts.ModuleResolutionKind} moduleResolution
+ * @returns {string | undefined}
+ */
+function normalizeModuleResolutionOption(moduleResolution) {
```

Comment:
This doesn't seem very future proof.

Is there a way to have TypeScript convert this for us?

Reply:
@styfle
I see.
I cloudn't find a way to get complete config objects by TypeScript, but I found [tsconfck](https://github.com/dominikg/tsconfck) from https://github.com/microsoft/TypeScript/issues/44573. 

Does this library look future proof, do you think? If you think it's OK, I will rewrite `src/utils/load-tsconfig-options.js`.

Reply:
I rewrited this & related parts by [5d85cd2](https://github.com/vercel/ncc/pull/914/commits/5d85cd27bd3637cef366f7840c0515f6ee3d5f69), which uses [tsconfck](https://github.com/dominikg/tsconfck).
Would be glad to review it.